File: Advanced Programming Methods/Toy Language/src/Model/PrintStatement.java
package Model;

import java.util.Collections;
import java.util.List;

//adds to the Out Table the integer result from the exp's evaluation
public class PrintStatement implements IStatement {
    private Expression exp;

    public PrintStatement(Expression exp){
        this.exp = exp;
    }
    @Override
    public ProgramState execute(ProgramState state) throws MyException {
        IList<Integer> out = state.getOut();
        IDictionary<String, Integer> symTbl = state.getSymTable();
        out.add(exp.eval(symTbl));
        return state;
    }
    public String toString(){
        return "print(" + exp.toString() + ")";
    }
}

File: Advanced Programming Methods/Toy Language C#/A2/Model/Statements/PrintStatement.cs
using System;
using A2.UI;
using A2.Model.Expressions;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A2.Model.Statements
{
    //adds to the Out Table the integer result from the exp's evaluation
    public class PrintStatement : IStatement{
        private Expression exp;

        public PrintStatement(Expression exp)
        {
            this.exp = exp;
        }
        public ProgramState execute(ProgramState state)
        {
            Data_Structures.IList<int> outList = state.getOut();
            Data_Structures.IDictionary<string, int> symTbl = state.getSymTable();

            outList.add(exp.eval(symTbl));
            return state;
        }
        public override string ToString()
        {
            return "print(" + exp.ToString() + ")";
        }
    }
}
